### PR TITLE
[#IOCOM-1022] Modified GetMessages to get has_preconditions flag from the new RCConfiguration model

### PR DESCRIPTION
**Pull request by michaeldisaro**
#### List of Changes
Created RCConfigurationUtility to implement some inner logic with injected components and RCConfigurationModel.
Refactored function and tests to user RCConfigurationUtility based on the new RCConfigurationModel instead of old utils logics based on previous configurations.

#### Motivation and Context
We need to migrate to the new model.

#### How Has This Been Tested?
* yarn build
* yarn test

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: GetMessages/getMessagesFunctions/getMessages.fallback.ts [hunk 1]
```diff
@@ -118,9 +110,7 @@ export const enrichContentData = (
   context: Context,
   messageModel: MessageModel,
   blobService: BlobService,
-  redisClient: RedisClient,
-  remoteContentConfigurationModel: RemoteContentConfigurationModel,
-  remoteContentConfigurationTtl: NonNegativeInteger,
+  rcConfigurationUtility: RCConfigurationUtility,
   categoryFetcher: ThirdPartyDataWithCategoryFetcher
   // eslint-disable-next-line max-params
```
Comment (Garma00) [inline GetMessages/getMessagesFunctions/getMessages.fallback.ts:115]: maybe we can remove this disable now

Comment (michaeldisaro) [reply GetMessages/getMessagesFunctions/getMessages.fallback.ts:115]: done in https://github.com/pagopa/io-functions-app-messages/pull/51/commits/c539e77784d456bf922ab57be29284a8e67f4ba7

**Review by Garma00 — DISMISSED** [review]

File: utils/remoteContentConfig.ts [hunk 2]
```diff
@@ -7,145 +8,97 @@ import { parse } from "fp-ts/lib/Json";
...
-        ),
-        E.chain(
-          flow(
-            RetrievedRCConfiguration.decode,
-            E.mapLeft(
-              () =>
-                new Error(
-                  "Cannot decode Remote Content Configuration Json from Redis"
-                )
+  public readonly getOrCacheMaybeRCConfigurationById = (
```
Comment (lucacavallaro) [inline utils/remoteContentConfig.ts:70]: Can you add a comment that describes what this method do?

Comment (michaeldisaro) [reply utils/remoteContentConfig.ts:70]: done in https://github.com/pagopa/io-functions-app-messages/pull/51/commits/c13e877eb91963770b1631b027fb1ad51972cf10

**Review by Garma00 — DISMISSED** [review]

**Review by acuiuli — DISMISSED** [review]
LGTM

**Review by Garma00 — APPROVED** [review]

**Pull request merged**